src/modules/75-cf/pages/feature-flags/components/NoFeatureFlags.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { ReactNode } from 'react'
+import { ButtonProps, Container, FontVariation, Text } from '@harness/uicore'
+import FlagDialog from '@cf/components/CreateFlagDialog/FlagDialog'
+import { NoData } from '@cf/components/NoData/NoData'
+import { useStrings } from 'framework/strings'
+import noFlagsImg from '@cf/images/Feature_Flags_Teepee.svg'
+import noResultsImg from '@cf/images/EmptySearchResults.svg'
+
+export interface NoFeatureFlagsProps {
+  hasFeatureFlags: boolean
+  hasSearchTerm: boolean
+  hasFlagFilter: boolean
+  environmentIdentifier: string
+  clearFilter: () => void
+  clearSearch: () => void
+}
+
+export const NoFeatureFlags: React.FC<NoFeatureFlagsProps> = ({
+  hasFeatureFlags,
+  hasSearchTerm,
+  hasFlagFilter,
+  environmentIdentifier,
+  clearFilter,
+  clearSearch
+}) => {
+  const { getString } = useStrings()
+
+  const mainMessage = (): string => {
+    if (hasFeatureFlags) {
+      if (hasFlagFilter) return getString('common.filters.noMatchingFilterData')
+      if (hasSearchTerm) return getString('cf.noResultMatch')
+    }
+    return getString('cf.noFlag')
+  }
+
+  const buttonText = (): string => {
+    if (hasFlagFilter) return getString('cf.featureFlags.resetFilters')
+    if (hasSearchTerm) return getString('cf.featureFlags.clearSearch')
+    return ''
+  }
+
+  const buttonProps = (): ButtonProps => {
+    if (hasFlagFilter) {
+      return {
+        text: getString('cf.featureFlags.resetFilters'),
+        icon: 'reset',
+        minimal: true,
+        onClick: clearFilter
+      }
+    }
+    if (hasSearchTerm) {
+      return {
+        text: getString('cf.featureFlags.clearSearch'),
+        icon: 'x',
+        iconProps: { size: 10 },
+        minimal: true,
+        onClick: clearSearch
+      }
+    }
+    return {}
+  }
+
+  const additionalContent = (): ReactNode => {
+    if (hasFeatureFlags && hasFlagFilter) {
+      return <Text font={{ variation: FontVariation.BODY1 }}>{getString('cf.featureFlags.changeOrReset')}</Text>
+    }
+    if (!hasFeatureFlags) {
+      return <FlagDialog environment={environmentIdentifier} />
+    }
+  }
+
+  return (
+    <Container height="100%" flex={{ justifyContent: 'center', alignItems: 'flex-start' }} padding="xxxlarge">
```

Comment:
Why use `flex-start`? Would it not be better to align to the centre of the available space like other no data usage? This with the `xxxlarge` padding here and on the `NoData` could cause unnecessary overflows on smaller screen sizes.
```suggestion
    <Container height="100%" flex={{ align: 'center-center' }}>
```

Reply:
With the filter cards added, the image and especially 'Clear Filter/Search/New Flag' buttons are pushed quite far down the screen when viewing on smaller laptop & tablet sizes so I thought this looked better.  If your screen is narrow enough that the filter cards stack, this becomes even worse in that you must scroll to use the buttons
<img width="1369" alt="image" src="https://user-images.githubusercontent.com/5584348/175284694-b31db438-199e-4288-a97e-6191fc92c07e.png">
<img width="1542" alt="image" src="https://user-images.githubusercontent.com/5584348/175284824-b07fb2f0-d295-4ff9-9a1a-c5da2e116cdf.png">

Will I catch up with @Kseniia-Dudina and see what she thinks?